src/scenes/Event/scenes/oracle.js

```diff
@@ -182,6 +165,39 @@ class OraclePage extends React.Component {
     ];
   }
 
+  isActionButtonDisabled() {
+    const {
+      currentOptionIdx,
+      voteAmount,
+      config,
+      transactions,
+      address,
+    } = this.state;
+
+    // Not within correct times or not the result setter
+    if (config.predictionAction.btnDisabled) {
+      return true;
+    }
+
+    // Already have a pending tx for this Oracle
+    const pendingTxs = _.filter(transactions, { oracleAddress: address, status: TransactionStatus.Pending });
+    if (pendingTxs > 0) {
+      return true;
+    }
+
+    // Did not select an option
+    if (currentOptionIdx === -1) {
+      return true;
+    }
+
+    // Did not enter an amount
+    if ((voteAmount === 0 || Number.isNaN(voteAmount)) && config.predictionAction.showAmountInput) {
+      return true;
+    }
+
```

Comment:
also we will want to handle the set/finalize stage. If a result is already set. We should allow another set. If the event is already finalized, we shouldn't allow another finalize, and should go to topic page.

Reply:
@liviavinci  if a result is already set, we shouldn't allow another set

Reply:
ditto. can only setResult once. regarding the finalize route to topic page, we have another issue open for this. will fix in that other PR.